### PR TITLE
Update app to Ember 3

### PR DESCRIPTION
**Pull request by lfarrell**
https://jira.lib.unc.edu/browse/BXC-2676

### CODE REVIEW

File: client/app/utils/value-entry.js [hunk 1]
```diff
@@ -11,21 +11,23 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-import Ember from 'ember';
+import EmberObject from '@ember/object';
+import { computed } from '@ember/object';
+import { isEmpty } from '@ember/utils';
 
-const EMAIL_REGEXP = /^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;
+const EMAIL_REGEXP = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;
```
Comment (bbpennel) [inline client/app/utils/value-entry.js:18]: The `/` in this regex doesn't need to be escaped anymore?

File: client/package.json [hunk 2]
```diff
@@ -1,56 +1,69 @@
...
+    "jshint": "^2.12.0",
+    "loader.js": "^4.7.0",
+    "morgan": "^1.10.0",
+    "npm-run-all": "^4.1.5",
+    "qunit-dom": "^1.2.0"
   },
-  "repository": "",
   "engines": {
-    "node": ">= 0.12.0"
+    "node": "10.* || >= 12"
```
Comment (bbpennel) [inline client/package.json:61]: If i'm reading this correctly its allowing node 10 or 12+? Are we actually checking that 10 works?

File: .travis.yml [hunk 3]
```diff
@@ -2,22 +2,19 @@ sudo: false
...
-    - export PATH=/usr/local/phantomjs-2.0.0/bin:$PATH
-    # starting a GUI to run tests, per https://docs.travis-ci.com/user/gui-and-headless-browsers/#Using-xvfb-to-Run-Tests-That-Require-a-GUI
-    - export DISPLAY=:99.0
-    - sh -e /etc/init.d/xvfb start
     - "npm config set spin false"
-    - "npm install -g npm@^2"
+    - "npm install -g npm@^6"
 install:
     - npm install babel-plugin-transform-es2015-block-scoping
     - npm install -g bower
```
Comment (bbpennel) [inline .travis.yml:20]: With bower gone, should this be removed?

**Review by bbpennel — CHANGES_REQUESTED** [review]
Haven't tested yet, but not much jumped out at me as problematic. Added a few comments, but good work overall

File: client/app/utils/value-entry.js (hunk 1 above)
Comment (lfarrell) [reply client/app/utils/value-entry.js:18]: The style check gave an error about unneeded escaping

File: client/package.json (hunk 2 above)
Comment (lfarrell) [reply client/package.json:61]: No. That got auto updated. I can switch it to just >= 12

File: client/app/utils/value-entry.js [hunk 4]
```diff
@@ -19,13 +19,13 @@ const DATE_MONTH_REGEXP = /^[1-2]\d{3}-(0[1-9]|1[0-2])$/;
 const DATE_YEAR_REGEXP = /^[1-2]\d{3}$/;
 const DATE_ADMIN_REGEXP = /^([1-2]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[1-2]\d|3[0-1])|[1-2]\d{3}-(0[1-9]|1[0-2])|^[1-2]\d{3}$)$/;
 const DURATION_REGEXP = /^P.+$/;
-const ORCID_REGEXP = /^((https?:\/\/)?orcid\.org\/)?\d{4}-\d{4}-\d{4}-\d{4}$/;
+const ORCID_REGEXP = /^((https?:\/\/)?orcid\.org\/)\d{4}-\d{4}-\d{4}-\d{4}$/;
```
[the remark this answers is not in the document]
Comment (lfarrell) [reply client/app/utils/value-entry.js:24]: Don't remember why I removed that. I'll check. Seems like one of the tests failed if it was there.

[the remark this answers is not in the document]
Comment (lfarrell) [reply client/app/utils/value-entry.js]: Weird, with the ? in the regex the test for a disallowed orcid should have allowed the test orcid. Not sure how the test ever passed.

**Review by bbpennel — APPROVED** [review]

**Pull request merged**